x-pack/plugins/ml/server/models/job_service/jobs.js

```diff
@@ -322,6 +323,42 @@ export function jobsProvider(callWithRequest) {
     return { jobIds };
   }
 
+  // Checks if each of the jobs in the specified list of IDs exist.
+  // Job IDs in supplied array may contain wildcard '*' characters
+  // e.g. *_low_request_rate_ecs
+  async function jobsExist(jobIds = []) {
+    // Get the list of job IDs.
+    // Use size of 10000, matching anomaly_detectors endpoint.
+    const maxJobsSize = 10000;
+    const resp = await callWithRequest('search', {
+      index: ML_CONFIG_INDEX_PATTERN,
+      size: maxJobsSize,
+      body: {
+        _source: 'job_id',
+        query: {
+          term: { job_type: 'anomaly_detector' }
+        }
+      }
+    });
+
+    const results = {};
+    let allJobIds = [];
+    if (resp.hits.total.value > 0) {
+      const hits = resp.hits.hits;
+      allJobIds = hits.map(hit => hit._source.job_id);
+
+      // Check if each of the supplied IDs match existing jobs.
+      jobIds.forEach((jobId) => {
+        // Create a Regex for each supplied ID as wildcard * is allowed.
+        const regexp = new RegExp(`^${jobId.split(/\*+/).join('.*')}$`);
```

Comment:
is this replacing all instances of `*` with `.*`?
If so, I wonder if a global replace on the id would make it a bit more readable.
something like
```
const regexp = new RegExp(`^${jobId.replace(/\*+/g, '.*')}$`);
```

Reply:
Good spot. I'll go with your suggestion.